src/s11/handlers/s11_CS_resp_handler.c

```diff
@@ -53,52 +53,52 @@ s11_CS_resp_handler(MsgBuffer* message, GtpV2MessageHeader* hdr, uint32_t sgw_ip
+    csr_info.status = msgData.cause.causeValue;
+	csr_info.s11_sgw_fteid.header.iface_type = 11;
+	csr_info.s11_sgw_fteid.header.teid_gre = msgData.senderFTeidForControlPlane.teidGreKey;
+	csr_info.s11_sgw_fteid.header.v4 = 1;
+	csr_info.s11_sgw_fteid.ip.ipv4.s_addr = msgData.senderFTeidForControlPlane.ipV4Address.ipValue;
+
+	csr_info.s5s8_pgwc_fteid.header.iface_type = 7;
+	csr_info.s5s8_pgwc_fteid.header.teid_gre = msgData.pgwS5S8S2bFTeid.teidGreKey;
+	csr_info.s5s8_pgwc_fteid.header.v4 = 1;
+	csr_info.s5s8_pgwc_fteid.ip.ipv4.s_addr = msgData.pgwS5S8S2bFTeid.ipV4Address.ipValue;
+
+	csr_info.s1u_sgw_fteid.header.iface_type = 1;
+	csr_info.s1u_sgw_fteid.header.teid_gre = msgData.bearerContextsCreated[0].s1USgwFTeid.teidGreKey;
+	csr_info.s1u_sgw_fteid.header.v4 = 1;
+	csr_info.s1u_sgw_fteid.ip.ipv4.s_addr = msgData.bearerContextsCreated[0].s1USgwFTeid.ipV4Address.ipValue;
+
+	csr_info.s5s8_pgwu_fteid.header.iface_type = 5;
+	csr_info.s5s8_pgwu_fteid.header.teid_gre = msgData.bearerContextsCreated[0].s5S8UPgwFTeid.teidGreKey;
+	csr_info.s5s8_pgwu_fteid.header.v4 = 1;
+	csr_info.s5s8_pgwu_fteid.ip.ipv4.s_addr = msgData.bearerContextsCreated[0].s5S8UPgwFTeid.ipV4Address.ipValue;
+
+	csr_info.pdn_addr.pdn_type = 1;
+	csr_info.pdn_addr.ip_type.ipv4.s_addr = msgData.pdnAddressAllocation.ipV4Address.ipValue;
 
 	
-	csr_info.msg_data.csr_Q_msg_m.pco_length = 0; 
+	csr_info.pco_length = 0;
 	if(msgData.protocolConfigurationOptionsIePresent == true)
 	{
-		csr_info.msg_data.csr_Q_msg_m.pco_length = msgData.protocolConfigurationOptions.pcoValue.count; 
-		memcpy(&csr_info.msg_data.csr_Q_msg_m.pco_options[0], &msgData.protocolConfigurationOptions.pcoValue.values[0], msgData.protocolConfigurationOptions.pcoValue.count);
+		csr_info.pco_length = msgData.protocolConfigurationOptions.pcoValue.count;
+		memcpy(&csr_info.pco_options[0], &msgData.protocolConfigurationOptions.pcoValue.values[0], msgData.protocolConfigurationOptions.pcoValue.count);
 	}
 	else
 	{
 		/* Temporary hardcoding so that UE gets min DNS address.*/
 		char pco_options[27] = {0x80, 0x80, 0x21, 0x10, 0x03, 0x00, 0x00,0x10, 0x81, 0x06, 0x08,0x08,0x08, 0x08,0x83,0x06,0x08,0x08,0x08,0x04,0x00,0x0d, 0x04,0x08,0x08,0x08,0x08};
-		memcpy(&csr_info.msg_data.csr_Q_msg_m.pco_options[0], &pco_options[0], 27);
-		csr_info.msg_data.csr_Q_msg_m.pco_length = 27;
+		memcpy(&csr_info.pco_options[0], &pco_options[0], 27);
+		csr_info.pco_length = 27;
 	}
 
 
 
-	csr_info.destInstAddr = htonl(mmeAppInstanceNum_c);
-	csr_info.srcInstAddr = htonl(s11AppInstanceNum_c);
+	csr_info.header.destInstAddr = htonl(mmeAppInstanceNum_c);
+	csr_info.header.srcInstAddr = htonl(s11AppInstanceNum_c);
 
 	/*Send CS response msg*/
-	send_tipc_message(g_resp_fd, mmeAppInstanceNum_c, (char *)&csr_info, GTP_READ_MSG_BUF_SIZE);
+	send_tipc_message(g_resp_fd, mmeAppInstanceNum_c, (char *)&csr_info, sizeof(struct csr_Q_msg));
 	log_msg(LOG_INFO, "Send CS resp to mme-app stage6.\n");
```

Comment:
Move this log before sending tipc message.

Reply:
Done.